From efaf15f95ce1fb3f5e7181f4be7dd468351e2cbb Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 13 Dec 2023 14:28:22 +0000
Subject: [PATCH 0796/1085] dts: rp1: add SNPS quirk to USB3 host controllers

Set snps,parkmode-disable-ss-quirk for usb0 and usb1.

Enabling this test/debug feature seems to prevent controller lockups
with bidirectional SS bulk endpoints active.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/rp1.dtsi | 2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm/boot/dts/broadcom/rp1.dtsi
@@ -1017,6 +1017,7 @@
 			usb3-lpm-capable;
 			snps,axi-pipe-limit = /bits/ 8 <8>;
 			snps,dis_rxdet_inp3_quirk;
+			snps,parkmode-disable-ss-quirk;
 			snps,tx-max-burst-prd = <8>;
 			snps,tx-thr-num-pkt-prd = <2>;
 			interrupts = <RP1_INT_USBHOST0_0 IRQ_TYPE_EDGE_RISING>;
@@ -1030,6 +1031,7 @@
 			usb3-lpm-capable;
 			snps,axi-pipe-limit = /bits/ 8 <8>;
 			snps,dis_rxdet_inp3_quirk;
+			snps,parkmode-disable-ss-quirk;
 			snps,tx-max-burst-prd = <8>;
 			snps,tx-thr-num-pkt-prd = <2>;
 			interrupts = <RP1_INT_USBHOST1_0 IRQ_TYPE_EDGE_RISING>;
